### DISCUSSION TITLE
Can PYSR be used to find a system of equations in which variables are coupled?

### DISCUSSION THREAD

**Discussion by reactingflow** [MilesCranmer/PySR#742] [Q&A]
Hello everyone,

I am new to PYSR. I've been tasked with analyzing a complex system where the variables are interdependent, meaning changes in one variable affect others in a coupled manner.

My question is: Can PYSR be used to derive a system of equations that accurately describes such a coupled relationship between variables?

PYSR seems to be a powerful tool for symbolic regression, but as a first-time user, I'm unsure if it's capable of handling systems where multiple outputs depend on each other through a set of equations rather than just finding a single equation relating inputs to a single output.

Any advice or pointers on how to get started with this would be greatly appreciated.

Thank you in advance for your help!

**Comment by reactingflow** [comment]
Hello again,

I wanted to add some more context to my previous post regarding the nature of the system I am studying. For our research involving reaction and transport phenomena, the combinations of these coupled variables are actually limited. 

Would this information make it easier to apply PYSR, or perhaps suggest any specific strategies for using the software effectively in this scenario?

Thanks again for any insights you can offer.

Best regards,

**Comment by MilesCranmer** [comment]
I guess you could write a custom loss function to set this up although it will a bit complex. I would start by going through the examples on the docs and then seeing some of the other threads discussing custom loss functions

**Comment by mijanhuq** [comment]
Hello,

Following up on this thread. I am interested in trying to find a set of coupled ODE's using PySR. At present, it is possible to try to fit one equation at a time. 

Has there been any work since the inception of this post in trying to either learn two or more equations (whether ODEs or not)? 
Thanks in advance for any advice you may be able to give.

Regards,
Mijan

**Comment by MilesCranmer** [comment]
Yes! This is now totally possible using TemplateExpressionSpec. Some of the other recent discussions discuss highly relevant topics

**Reply by mijanhuq** [reply]
Thank you. I will look for them.
